src/read_pair_iter.rs

```diff
@@ -322,6 +322,36 @@ impl ReadPairIter {
         })
     }
 
+    pub fn is_single_ended(&self) -> Result<bool, FastqError> {
```

Comment:
This should return the value of a flag stored in the struct.  THere should probably be a non-public method that computes this once at the time you create the `ReadPairIter`

Reply:
Fixed.